app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductBrowsingListActivity.java

```diff
@@ -258,67 +260,73 @@ private void setUpRecyclerView() {
         offlineCloudLayout.setVisibility(View.INVISIBLE);
 
         productsRecyclerView.setVisibility(View.VISIBLE);
-        productsRecyclerView.setHasFixedSize(true);
-        LinearLayoutManager mLayoutManager = new LinearLayoutManager(ProductBrowsingListActivity.this, LinearLayoutManager.VERTICAL, false);
-        productsRecyclerView.setLayoutManager(mLayoutManager);
 
-        ProductsRecyclerViewAdapter adapter = new ProductsRecyclerViewAdapter(mProducts);
-        productsRecyclerView.setAdapter(adapter);
+        if (!setupDone) {
```

Comment:
Please ensure that this will not prevent the list from being used throughout the app lifecycle